src/ray/gcs/entry_change_notification.h

```diff
@@ -0,0 +1,51 @@
+#ifndef RAY_GCS_ENTRY_CHANGE_NOTIFICATION_H
+#define RAY_GCS_ENTRY_CHANGE_NOTIFICATION_H
+
+#include <ray/protobuf/gcs.pb.h>
+#include <vector>
+
+namespace ray {
+
+namespace gcs {
+
+/// \class EntryChangeNotification
+/// EntryChangeNotification class is a template class which represent
+/// notification of entry change from GCS.
+template <typename Data>
+class EntryChangeNotification {
+ public:
+  EntryChangeNotification(rpc::GcsChangeMode change_mode, std::vector<Data> data)
+      : change_mode_(change_mode), data_(std::move(data)) {}
+
+  EntryChangeNotification(EntryChangeNotification &&other) {
+    change_mode_ = other.change_mode_;
+    data_ = std::move(other.data_);
+  }
+
+  EntryChangeNotification &operator=(EntryChangeNotification &&other) {
+    change_mode_ = other.change_mode_;
+    data_ = std::move(other.data_);
+  }
+
+  /// Get change mode of this notification.
+  ///
+  /// \return rpc::GcsChangeMode
+  rpc::GcsChangeMode GetGcsChangeMode() const { return change_mode_; }
```

Comment:
Would be better to avoid expose `GcsChangeMode` in the API, which is an implementation detail. 
We can just expose `IsAdded` and `IsRemoved`.

Reply:
I don't think there will be other types. Even if a new type is added in the future, we can easily add this back. But for now, I'd  suggesting avoid exposing implementation details.

Reply:
Ok. Let keep this interface for test only.